src/interpreter/inner.rs

```diff
@@ -274,18 +308,46 @@ pub fn from_txdata<'txin>(
     // ** bare script **
     } else {
         if wit_stack.is_empty() {
-            let miniscript = Miniscript::<bitcoin::PublicKey, _>::parse_insane(spk)?;
+            // Bare script parsed in BareCtx
+            let miniscript = Miniscript::<bitcoin::PublicKey, BareCtx>::parse_insane(spk)?;
+            let miniscript = pre_taproot_to_no_checks(&miniscript);
             Ok((
                 Inner::Script(miniscript, ScriptType::Bare),
                 ssig_stack,
-                spk.clone(),
+                Some(spk.clone()),
             ))
         } else {
             Err(Error::NonEmptyWitness)
         }
     }
 }
 
+// Convert a miniscript from a specified context into an insane miniscript
+// We need to remember how the hash160 was translated because while doing a checksig
+// we need to know whether to parse the public key provided in witness as x-only or full
+pub(super) fn pre_taproot_to_no_checks<Ctx: ScriptContext>(
```

Comment:
this function / its name is kinda confusing; maybe you can clarify the comment to discuss what the goal is (parse the public key provided in witness as x-only or full) and then what we do & why it accomplishes that?

Maybe you can also add a crate local trait IsPreTaproot to restrict what Ctx it can be called with.

Reply:
or a trait crate local ToNoChecks and you can impl it default, and change just Tap CTX's impl?

Reply:
The interpreter API operates on a NoChecks context because making it context-specific would involve lots of duplicate code for each context. The interpreter is for the use case when you already have a transaction and want to figure out what constraints were satisfied. The primary use-case for this feature was miniscript friendly block explorers that would output what constraints were satisfied for this given transaction. Hence, the insane scripts. 

As for this PR, I have added a `ToNoChecksTrait` that makes the code look cleaner.